### PR TITLE
Filter out project names from potential user targets

### PR DESCRIPTION
**Pull request by cdmihai**

### CODE REVIEW

**Comment by jeffkl** [comment]
The point of this is to act as a pass through of targets from the solution to the project. If there are "invalid" target names and one is specified, we should probably error.

**Comment by rainersigwald** [comment]
They're not really "illegal" or "invalid", they're "already in use in the metaproj" and thus shouldn't be added (again, causing the error).

File: src/XMakeBuildEngine/Construction/Solution/SolutionProjectGenerator.cs [hunk 1]
```diff
@@ -58,7 +58,23 @@ internal class SolutionProjectGenerator
...
+            "Build",
+            "Clean",
+            "Rebuild",
+            "Publish"
+            );
+
+        /// <summary>
+        /// Internal implementation targets that can't be used by users
+        /// </summary>
+        private static readonly ISet<string> _illegaUserTargetNames = ImmutableHashSet.Create<string>(StringComparer.OrdinalIgnoreCase,
```
Comment (rainersigwald) [inline src/XMakeBuildEngine/Construction/Solution/SolutionProjectGenerator.cs]: rename `_illega` to `_solutionGeneratedTargetNames`

**Comment by cdmihai** [comment]
Replaced by #1545

**Pull request closed without merging**